biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -203,6 +258,84 @@ subroutine create_cohort(currentSite, patchptr, pft, nn, hite, dbh, bleaf, bfine
 
   end subroutine create_cohort
 
+  ! -------------------------------------------------------------------------------------
+
+  subroutine InitPRTCohort(new_cohort)
+
+     ! ----------------------------------------------------------------------------------
+     ! This subroutine simply allocates and attaches the correct PRT object.
+     ! The call to InitPRTVartype() performs the allocation of the variables
+     ! and boundary conditions inside the object.  It also initializes
+     ! all values as unitialized (large bogus values).
+     !
+     ! Each PARTEH allocation hypothesis has different expectations of boundary conditions.
+     ! These are specified by pointers to values in the host model. Because these
+     ! are pointers, they just need to be set once when the prt object is first initalized.
+     ! The calls below to "RegisterBCINOut", "RegisterBCIn" and "RegisterBCOut" are
+     ! setting those pointers.
+     ! -----------------------------------------------------------------------------------
+
+     !
+     ! !ARGUMENTS    
+     type(ed_cohort_type), intent(inout), target  :: new_cohort
+     type(callom_prt_vartypes), pointer :: callom_prt
+
+
+     ! Allocate the PRT class object
+     ! Each hypothesis has a different object which is an extension
+     ! of the base class.
+
+     select case(hlm_parteh_model)
+     case (prt_carbon_allom_hyp)
+        
+        allocate(callom_prt)
+        new_cohort%prt => callom_prt
+     
+     case DEFAULT
+
+        write(fates_log(),*) 'You specified an unknown PRT module'
+        write(fates_log(),*) 'Aborting'
+        call endrun(msg=errMsg(sourcefile, __LINE__))
+
+     end select
+     
+     ! This is the call to allocate the data structures in the PRT object
+     ! This call will be extended to each specific class.
+
+     call new_cohort%prt%InitPRTVartype()
+
+
+     ! Set the boundary conditions that flow in an out of the PARTEH
+     ! allocation hypotheses.  These are pointers in the PRT objects that
+     ! point to values outside in the FATES model.
+
+     ! Example:
+     ! "ac_bc_inout_id_dbh" is the unique integer that defines the object index
+     ! for the allometric carbon "ac" boundary condition "bc" for DBH "dbh"
+     ! that is classified as input and output "inout".
+     ! See PRTAllometricCarbonMod.F90 to track its usage.
+     ! bc_rval is used as the optional argument identifyer to specify a real
+     ! value boundary condition.
+     ! bc_ival is used as the optional argument identifyer to specify an integer
+     ! value boundary condition.
+     
+
+     select case(hlm_parteh_model)
+     case (prt_carbon_allom_hyp)
+
+        ! Register boundary conditions for the Carbon Only Allometric Hypothesis
```

Comment:
maybe clarify as
! Register boundary conditions (bc) for the Carbon Only Allometric Hypothesis (ac)

Reply:
sounds good, will do